### PR TITLE
Review in depth

### PR DESCRIPTION
**Pull request by felixpacheco**

### CODE REVIEW

File: tecan_od_analyzer/tecan_od_analyzer.py [hunk 1]
```diff
@@ -276,7 +274,7 @@ def compensation_lm(cor_df, df_gr) : #done
 	linear = lambda x, a, b: a * x + b
 	sns.set(style="white", palette="muted", color_codes=True)
 	fig = plt.figure()
-	fig.suptitle('Linear models of volume loss correlation to time for different plates', fontweight="bold")
+	fig.suptitle('Volume loss correlation over time', fontweight="bold")
 	#sns.despine(left=True)
```
Comment (dmccloskey) [inline tecan_od_analyzer/tecan_od_analyzer.py:278]: If this line is not needed, I would recommend removing it.  If it might be needed later at some point, I would recommend adding that into the comment.

File: tecan_od_analyzer/tecan_od_analyzer.py [hunk 2]
```diff
@@ -289,15 +287,13 @@ def compensation_lm(cor_df, df_gr) : #done
 		lm_eq.append(popt)
 		
 		ax = fig.add_subplot(math.ceil(len(unique_bioshaker)), math.ceil(len(unique_bioshaker)/2), shaker+1)
-		#ax.get_xaxis().set_visible(True)
-		#ax.get_yaxis().set_visible(True)
 		ax.plot(sub_cor_df["time_hours"], sub_cor_df["Correlation"], "o", label="Empirical data", markerfacecolor='skyblue', markeredgecolor="dodgerblue", alpha= 0.5)
 		ax.plot(sub_cor_df["time_hours"], linear(sub_cor_df ["time_hours"], *popt), "b-", label="Linear model", color = "darkred")
-		ax.set_ylabel('Volume loss correlation', labelpad=10)
+		ax.set_ylabel('Correlation', labelpad=10)
```
Comment (dmccloskey) [inline tecan_od_analyzer/tecan_od_analyzer.py:292]: In general, it is good to specify the units for all axes labels.  If the units are arbitrary, please us (au) for arbitrary units.  If units are not applicable, please use (na).

File: tests/test_tecan_od_analyzer.py [hunk 3]
```diff
@@ -219,7 +215,7 @@ def test_gr_estimation(self):
 		self.assertIsInstance(result2, pd.DataFrame)
 		self.assertIsInstance(result3, list)
 
-		#Length of outputs
+		# Test not lost samples
```
Comment (dmccloskey) [inline tests/test_tecan_od_analyzer.py:218]: What do you mean by "not lost samples"?

File: tests/test_tecan_od_analyzer.py [hunk 4]
```diff
@@ -246,10 +242,17 @@ def test_interpolation(self):
...
-		#Comparison to expected output
-		#self.assertTrue(expected_result["Estimation"].tolist()-0.01<=result["Estimation"].tolist()<= expected_result["Estimation"].tolist()+0.01)
+
+		expected_result = expected_result["Estimation"].tolist()
+		result = result["Estimation"].tolist()
+
+		for i in range(len(result)):
+			
+			#Comparison to expected output
+			self.assertTrue(result[i]-0.5<= expected_result[i]<= result[i]+0.5)
```
Comment (dmccloskey) [inline tests/test_tecan_od_analyzer.py:255]: The tolerance used seems to be quite a bit higher than what would be normally used for floating point number comparison.  Is there some sort of stochasticity in the calculations?

**Review by dmccloskey — CHANGES_REQUESTED** [review]
# Summary
Overall, I am quite pleased with the results of your work.  The scripts you have developed should provide us with a lot of value, and hopefully we will not have to contact you for any/too many questions.

## Positives
1. Docstrings are well done
2. Code quality looks quite high and is in general easy to read and understand
3. Documentation of the input is clear

## Major comments
1. Missing command line option for the input directory and the results directory
2. Missing units tests for `stats_summary`, `exponential`, and `interpolation`
3. Missing documentation for what the expected output file format should look like including an explanation of the different columns and units of the values in those columns

## Minor comments
1. Please check that all commented code is either removed if not necessary or an explanation is made as to why they are there
2. Please ensure that all diagnostic `print` messages that are not informative to the user are removed.
3. Please double check that all docstrings and unit tests are complete.  In particular, if there are any edge cases in the unit tests that are not tested for, but could be of concern, please do note them down in the tests and code for later developers to check.

**Comment by felixpacheco** [comment]
A new pull request will be created with the new features
